src/basic/fs-util.c

```diff
@@ -644,6 +644,20 @@ static bool safe_transition(const struct stat *a, const struct stat *b) {
         return a->st_uid == b->st_uid; /* Otherwise we need to stay within the same UID */
 }
 
+static int log_unsafe_transition(int a, int b, const char *path, unsigned flags) {
+        if (flags & CHASE_WARN) {
```

Comment:
can you invert this check and exit early? we try to optimize our code for early exit so that our indentation levels stay minimal